pkg/resource/clickpipe.go

```diff
@@ -1520,43 +1849,235 @@ func (c *ClickPipeResource) syncClickPipeState(ctx context.Context, state *model
 		sourceModel.Kinesis = types.ObjectNull(models.ClickPipeKinesisSourceModel{}.ObjectType().AttrTypes)
 	}
 
-	state.Source = sourceModel.ObjectValue()
+	if clickPipe.Source.Postgres != nil {
+		statePostgresModel := models.ClickPipePostgresSourceModel{}
+		var stateSettingsModel models.ClickPipePostgresSettingsModel
 
-	destinationModel := models.ClickPipeDestinationModel{
-		Database:        types.StringValue(clickPipe.Destination.Database),
-		Table:           types.StringValue(clickPipe.Destination.Table),
-		ManagedTable:    types.BoolValue(clickPipe.Destination.ManagedTable),
-		TableDefinition: types.Object{},
-		Columns:         types.List{},
-		Roles:           types.List{},
+		if !stateSourceModel.Postgres.IsNull() {
+			if diags := stateSourceModel.Postgres.As(ctx, &statePostgresModel, basetypes.ObjectAsOptions{}); diags.HasError() {
+				return fmt.Errorf("error reading ClickPipe Postgres source: %v", diags)
+			}
+			// Get the state settings model
+			if !statePostgresModel.Settings.IsNull() {
+				if diags := statePostgresModel.Settings.As(ctx, &stateSettingsModel, basetypes.ObjectAsOptions{}); diags.HasError() {
+					return fmt.Errorf("error reading ClickPipe Postgres settings: %v", diags)
+				}
+			}
+		}
+
+		// Settings - preserve null values from state for optional fields
+		settingsModel := models.ClickPipePostgresSettingsModel{
+			ReplicationMode: types.StringValue(clickPipe.Source.Postgres.Settings.ReplicationMode),
+		}
+
+		if clickPipe.Source.Postgres.Settings.SyncIntervalSeconds != nil {
+			settingsModel.SyncIntervalSeconds = types.Int64Value(int64(*clickPipe.Source.Postgres.Settings.SyncIntervalSeconds))
+		} else {
+			settingsModel.SyncIntervalSeconds = types.Int64Null()
+		}
+
+		if clickPipe.Source.Postgres.Settings.PullBatchSize != nil {
+			settingsModel.PullBatchSize = types.Int64Value(int64(*clickPipe.Source.Postgres.Settings.PullBatchSize))
+		} else {
+			settingsModel.PullBatchSize = types.Int64Null()
+		}
+
+		// For optional string fields, if state was null, keep it null (API may return empty string)
+		if stateSettingsModel.PublicationName.IsNull() {
+			settingsModel.PublicationName = types.StringNull()
+		} else if clickPipe.Source.Postgres.Settings.PublicationName != nil && *clickPipe.Source.Postgres.Settings.PublicationName != "" {
+			settingsModel.PublicationName = types.StringValue(*clickPipe.Source.Postgres.Settings.PublicationName)
+		} else {
+			settingsModel.PublicationName = stateSettingsModel.PublicationName
+		}
```

Comment:
Trying to wrap my head around this. If we're syncing the state from the API, wouldn't this be:
```go
if clickPipe.Source.Postgres.Settings.PublicationName != nil &&
   *clickPipe.Source.Postgres.Settings.PublicationName != "" {
    settingsModel.PublicationName = types.StringValue(*clickPipe.Source.Postgres.Settings.PublicationName)
} else {
    settingsModel.PublicationName = types.StringNull()
}
```
We're not _expecting_ desyncs but if the local state got corrupted and wasn't null but became null, isn't the sync's job to bring it back from null

Reply:
If this makes sense, should we explicitly map just (API default, state null) to (new state null) and not (API any, state null)?

Reply:
Yeah I think I had this backwards, switched the mapping now.